src/v/storage/segment_set.cc

```diff
@@ -197,6 +199,83 @@ is_last_segment(segment* s, std::optional<ss::sstring> last_clean_segment) {
                 == std::string(last_clean_segment.value());
 }
 
+ss::future<std::optional<segment_set>>
+maybe_create_contiguous_segment_set(segment_set::underlying_t segs) {
+    if (segs.empty()) {
+        co_return std::nullopt;
+    }
+    segment_set seg_set = segment_set(std::move(segs));
+    auto min_offset = seg_set.front()->offsets().get_base_offset();
+    // Max offset may not necessarily be seg_set.back()'s dirty_offset, due to
+    // descending ordering w/r/t dirty offset in `segment_ordering`.
+    auto max_offset = (*std::ranges::max_element(
+                         seg_set,
+                         std::less<>{},
+                         [](const auto& s) {
+                             return s->offsets().get_dirty_offset();
+                         }))
+                        ->offsets()
+                        .get_dirty_offset();
+    segment_set::underlying_t ignored_segs;
+    for (auto it = seg_set.begin(); it != seg_set.end();) {
+        if (it != seg_set.begin()) {
+            auto& s = *it;
+            auto& prev = *std::prev(it);
+            if (
```

Comment:
Given we're relying on ordering of the underlying segment_set, could we create a std::deque of segments in this method and sort them exactly how we need, rather than changing the sort order of segment_set and relying on it implicitly

If you choose to keep it as is, could you add a comment explaining that we are relying on a specific ordering of the last offsets of these segments?